File: articles/organizations-as-complex-behavioral-systems/go/complex_system_score.go
// Organizations as Complex Behavioral Systems
// Simple Go scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, institutional learning,
// and reproducible workflows only. Not for employment, productivity ranking,
// loyalty scoring, dissent tracking, or individual decisions.

package main

import "fmt"

type ComplexSystemProfile struct {
	PsychologicalSafety      float64
	LeadershipTrust         float64
	CulturalCoherence       float64
	IncentiveAlignment      float64
	InformationFlow         float64
	LearningCapacity        float64
	RedundancySlack         float64
	WorkloadPressure        float64
	SilenceRisk             float64
	FragmentationPressure   float64
	EnvironmentalUncertainty float64
}

func ComplexSystemRiskScore(p ComplexSystemProfile) float64 {
	return 0.11*(100.0-p.PsychologicalSafety) +
		0.11*(100.0-p.LeadershipTrust) +
		0.09*(100.0-p.CulturalCoherence) +
		0.09*(100.0-p.IncentiveAlignment) +
		0.09*(100.0-p.InformationFlow) +
		0.10*(100.0-p.LearningCapacity) +
		0.08*(100.0-p.RedundancySlack) +
		0.09*p.WorkloadPressure +
		0.08*p.SilenceRisk +
		0.10*p.FragmentationPressure +
		0.08*p.EnvironmentalUncertainty
}

func main() {
	profile := ComplexSystemProfile{84, 85, 82, 80, 83, 82, 68, 30, 18, 22, 42}
	fmt.Printf("Synthetic complex system risk score: %.2f\n", ComplexSystemRiskScore(profile))
	fmt.Println("Responsible-use reminder: institutional learning only.")
}

File: articles/organizations-as-complex-behavioral-systems/rust/complex_system_score.rs
// Organizations as Complex Behavioral Systems
// Simple Rust scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, institutional learning,
// and reproducible workflows only. Not for employment, productivity ranking,
// loyalty scoring, dissent tracking, or individual decisions.

struct ComplexSystemProfile {
    psychological_safety: f64,
    leadership_trust: f64,
    cultural_coherence: f64,
    incentive_alignment: f64,
    information_flow: f64,
    learning_capacity: f64,
    redundancy_slack: f64,
    workload_pressure: f64,
    silence_risk: f64,
    fragmentation_pressure: f64,
    environmental_uncertainty: f64,
}

fn complex_system_risk_score(p: &ComplexSystemProfile) -> f64 {
    0.11 * (100.0 - p.psychological_safety)
        + 0.11 * (100.0 - p.leadership_trust)
        + 0.09 * (100.0 - p.cultural_coherence)
        + 0.09 * (100.0 - p.incentive_alignment)
        + 0.09 * (100.0 - p.information_flow)
        + 0.10 * (100.0 - p.learning_capacity)
        + 0.08 * (100.0 - p.redundancy_slack)
        + 0.09 * p.workload_pressure
        + 0.08 * p.silence_risk
        + 0.10 * p.fragmentation_pressure
        + 0.08 * p.environmental_uncertainty
}

fn main() {
    let profile = ComplexSystemProfile {
        psychological_safety: 84.0,
        leadership_trust: 85.0,
        cultural_coherence: 82.0,
        incentive_alignment: 80.0,
        information_flow: 83.0,
        learning_capacity: 82.0,
        redundancy_slack: 68.0,
        workload_pressure: 30.0,
        silence_risk: 18.0,
        fragmentation_pressure: 22.0,
        environmental_uncertainty: 42.0,
    };

    println!(
        "Synthetic complex system risk score: {:.2}",
        complex_system_risk_score(&profile)
    );
    println!("Responsible-use reminder: institutional learning only.");
}
